pkgs/by-name/me/metal-stack-metalctl/package.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, makeWrapper
+, go-licenses
+, stdenv
+, glibc
+}:
+
+buildGoModule rec {
+  pname = "metal-stack-metalctl";
+
+  version = "0.16.2";
+  gitversion = "tags/v0.16.2-0-gf9c3071";  # git describe --long --all
+  gitsha = "f9c3071b";                     # git rev-parse --short=8 HEAD
+  gittime = "2024-05-17T19:05:41+02:00";   # date --iso-8601=seconds
```

Comment:
Let's migrate this.

```suggestion
```

Reply:
How can I get 1 patch for all the suggested changes so that I can test it?

And besides - why do you suggest this, what is wrong with the nix file? It builds and all tests ran ok.

Thanks, Tom

Reply:
> How can I get 1 patch for all the suggested changes so that I can test it?

I do not know how to use the GitHub interface. I merely copy-paste the things, it is easier for me.

> And besides - why do you suggest this, what is wrong with the nix file? It builds and all tests ran ok.

"Builds and runs OK" is not the litmus test.

The attributes you created here are being used only at the ldflags definition.
It is a good practice to put the variables in the most restrictive scopes.

I don't know if you have read old manuals about avoiding global variables in structured programming. The idea here is the same: putting an attribute in a scope too far can make it interact in unexpected ways.
Also it gives a visual clue: the attributes are near of the code that references them.

Reply:
Well, I don't know this as well. At least from what Github shows me, I can only see a couple of independent visual diffs. I can commit each or only a subset, but that's it. It does not provide a single patch which I can apply to my source to test it.

And I don't understand why there are 6 different diffs in the first place, as these obviously are connected to each other. I can't even put them into a patch file and apply this using `patch` because `patch` needs context.

So I have to manually edit the source remove/change the code line by line. Yes, this is just a small file, so I can do it. But with a larger file I'd just recject this. Nobody applies patches manually by editing a file. So, for the future, if you want to suggest connected changes to a PR author, please do it with ONE diff. Thank you.

Reply:
Ok, I have applied the patches. Please check.